### PR TITLE
API template updates

### LINKED ISSUE #49047: Multiple API template entry points should be reconciled
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In the VS new project dialog there is now multiple 'API' template options.  The descriptions do not disambiguate enough IMO of what they are and the one with a more complete description and tags seems to be the one I want and not the new one from .NET 8 SDK.  These should be reconciled with better descriptions:

image

Similarly:
image

### Expected Behavior

Better descriptions between the two and proper ordering to where you want them positioned in the dialog/search.

### Steps To Reproduce

1) Install latest .NET 8 SDK in an installation of VS
2) File...New Project

### Exceptions (if any)

_No response_

### .NET Version

_No response_

### Anything else?

_No response_

### LINKED ISSUE #49217: WebApi template remove UseMinimalAPIs option
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

The UseMinimalAPI option is duplicated with the Api template

### Describe the solution you'd like

remove UseMinimalAPIs option

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DamianEdwards**
Fixes #49047
Fixes #49217

In Visual Studio under the "API" project type:

image

In Visual Studio under the "Web API" project type:

image

At cmd line with `dotnet new list`:

image

### CODE REVIEW

**Comment by DamianEdwards** [comment]
@Varorbc it's more conservative than what was discussed but it's a PR so can still change :smile:

I'm still a bit concerned about the functionality take-back of the new template regarding OpenAPI and authentication options if we remove the minimal API option, and the truth is we're limited here by what's currently supported in the VS new project experience. In short, I don't feel like there's a good way to simply differentiate the two templates via their name as the feature-sets just don't compose well given the constraints.

**Comment by ilmax** [comment]
Just curious but what's the reasoning behind the "RESTful Web API" and "RESTful HTTP Api"?

I feel like those description could be aligned.

**Comment by DamianEdwards** [comment]
@ilmax see #49217 but I'm updating this now for [latest proposal](https://github.com/dotnet/aspnetcore/issues/49217#issuecomment-1638723166) which will align the names along a different pivot

**Comment by DamianEdwards** [comment]
@Varorbc those tags in VS actually don't come from the template definition. There will be a set of changes to assets in VS itself to add those aspects.

**Review by mitchdenny — APPROVED** [review]

File: src/ProjectTemplates/Web.ProjectTemplates/Api-CSharp.csproj.in [hunk 1]
```diff
@@ -6,10 +6,8 @@
     <ImplicitUsings>enable</ImplicitUsings>
```
[the remark this answers is not in the document]
Comment (DamianEdwards) [reply src/ProjectTemplates/Web.ProjectTemplates/Api-CSharp.csproj.in:6]: We previously haven't renamed template dirs & files when the names change (you can see this from some of the other names) so I didn't do it this time either. That said, I'm not sure there's any good reason why not and am I happy to send a follow-up PR with that change.

**Pull request merged**